### ISSUE TITLE
Feature request

### ISSUE THREAD

**Issue by Hellowlol** [idlesign/torrentool#5]
- Support for webseed and http seed
- Dynamically set a piece size
- Set max piece size (some torrent sites dont like huge torrents)

**Comment by idlesign** [comment]
Thank you. 
Sounds good, yet this should rather be three separate requests, so we trace all of them down one be one.

**Comment by Hellowlol** [comment]
some ideas on how to calc piece size etc.
https://repl.it/DN1e

**Comment by idlesign** [comment]
I see no reason for a user to set pieces number as setting it involves manual size calculation from his side to devise a "proper" number.

On the other hand setting size (in KiBs) manually is probably a good thing.

As a whole, I think we better stick to some widely-used predefined values for various file sizes (something like mode `auto` in your example).

Some useful links:
* https://github.com/wiiiky/libtransmission/blob/master/src/makemeta.c#L89 (`Transamission` calculation).
* https://wiki.theory.org/BitTorrentSpecification (**Notes** section under `Metainfo File Structure`)
* http://wiki.vuze.com/w/Torrent_Piece_Size
* https://torrentfreak.com/how-to-make-the-best-torrents-081121/ (under `PIECE SIZE` section)

**Comment by Hellowlol** [comment]
The reason i added this was because some torrent sites have a max size for the torrent files. Reducing number of pieces reduces the torrent size. The calc mode is what deluge uses. I think beta is the best.

**Comment by idlesign** [comment]
> [...] because some torrent sites have a max size for the torrent files [...]

Yes, but the same goal is already achieved with piece size option. How do you expect a user to set pieces number to control .torrent file size without some precalculations on his side?

Maybe we need an option named like `target .torrent size` to deduce piece size and number from, but this will involve some heuristics.

`beta` pushes it too hard — to many small pieces on small torrents, and too big chunks on large, both will affect preformance/availability. Also it recalculates sizes every time, and there is no reason to do it, since file size ranges will do just fine. Any particular reason you consider `beta` the best?

**Comment by Hellowlol** [comment]
Because of the pieces will be in the range of 1200 and 1800. Unless the file is to small to get the desired pieces or the file is simply to large to get the pieces it want with max piecesize. And it want smaller pieces size you can set a new max piece size, 16 mb might be over the top :p 

It's the follows torrent freaks guide the closest. Seeding a 200-300 gb torrent (not uncommon on tv torrent sites) with 1 mb piece size result in a huge overhead.

**Comment by idlesign** [comment]
Chunks under 32KiB are not practical considering current bandwidths, and over 16MiB are too havy load anyway.
Let's base on `Transmission` presets but extend them up to 16MiB, and let's give an option to set actual size. Later we shuold try to create a machinery handling `max-torrent-size` option, so as to be more user-friendly.

**Comment by Hellowlol** [comment]
I have started on some of the changes i have listed. It needs a serious cleanup and more tests but it shows the direction i want to go. Ill send a pr when im done and you can just pick what you need or reject the entire thing :P, I just wanted to give you a heads up. https://github.com/Hellowlol/torrentool/tree/better_plx

**Comment by idlesign** [comment]
Ok, I'll watching from time to time.

**Comment by modbender** [comment]
@idlesign Also please make it so that all trackers can be included in the magnet url, sometimes it might turn out that the primary tracker won't work because file is removed, then it won't download.
